src/Security/samples/CustomPolicyProvider/Authorization/MinimumValueAuthorizationHandler.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Routing;
+
+namespace CustomPolicyProvider
+{
+    internal class MinimumValueAuthorizationHandler : AuthorizationHandler<MinimumValueAuthorizationRequirement>
+    {
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, MinimumValueAuthorizationRequirement requirement)
```

Comment:
Adding the `HttpContext` to `AuthorizationHandlerContext` is a no-go due to layering: https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authorization/Core/src/Microsoft.AspNetCore.Authorization.csproj#L17-L21. 

This change introduces a new context type that can be accessed as the "resource" which exposes the request context. Here's how you configure it: https://github.com/aspnet/AspNetCore/pull/16714/files#diff-5350d82af40fc53aa98d53ca127bd589R25